code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -750,6 +750,111 @@
 	var/obj/item/pushbroom/cyborg/BR = locate() in R.model.modules
 	if (BR)
 		R.model.remove_module(BR, TRUE)
+		
+/obj/item/borg/upgrade/condimentsynthesizer
+	name = "Service Cyborg Condiment Synthesiser"
+	desc = "An upgrade to the service model cyborg, allowing it \
+		to produce solid condiments."
+	icon_state = "cyborg_upgrade3"
+	require_model = TRUE
+	model_type = list(/obj/item/robot_model/service)
+	model_flags = BORG_MODEL_SERVICE
+
+/obj/item/borg/upgrade/condimentsynthesizer/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/reagent_containers/borghypo/condimentsynthesizer/CS = locate() in R.model.modules
```

Comment:
don't do abbreviation vars in new code, we have to inherit `action` having the arg `R` unfortunately, but we shouldn't repeat the pattern in modern code. Change it to something like

```suggestion
		var/obj/item/reagent_containers/borghypo/condimentsynthesizer/synth = locate() in R.model.modules
```

or whatever you want really. abbreviations are just wack and this is no-where near being hot-enough to necessitate having abbreviations for var names.